tools/offcputime.py

```diff
@@ -172,25 +208,35 @@ def signal_ignore(signal, frame):
 stack_traces = b.get_table("stack_traces")
 for k, v in sorted(counts.items(), key=lambda counts: counts[1].value):
     # handle get_stackid erorrs
-    if k.stack_id < 0:
+    if (not args.user_stacks_only and k.kernel_stack_id < 0) or \
+            (not args.kernel_stacks_only and k.user_stack_id < 0 and \
+                k.user_stack_id != -14):
         missing_stacks += 1
         # check for an ENOMEM error
-        if k.stack_id == -12:
+        if k.kernel_stack_id == -12 or k.user_stack_id == -12:
```

Comment:
You could use the `errno` module to drop the magic numbers:

    if k.kernel_stack_id == -errno.ENOMEM or k.user_stack_id == -errno.ENOMEM: